shell/middleware/authenticated.js

```diff
@@ -71,9 +71,14 @@ export function getProductFromRoute(to) {
   return product;
 }
 
-function setProduct(store, to) {
+function setProduct(store, to, redirect) {
```

Comment:
Cannot do that, otherwise we won't stop the execution of the `authenticated` middleware with the redirect. Putting it inside the `setProduct` function will only "return" it as an output of the function.

Reply:
Not sure i get that.

```
store.dispatch('loadingError', new Error(store.getters['i18n/t']('nav.failWhale.productNotFound', { productNotFound }, true)));

return redirect(302, '/fail-whale');
```
This can move in to `setProduct`, only called if needed. 

Then

```
const redirected = setProduct(store, to)
if (redirected) {
  return redirected
}
```